src/test/lib/sql/sql_pipeline_statement_test.cpp

```diff
@@ -91,6 +91,8 @@ class SQLPipelineStatementTest : public BaseTest {
   std::shared_ptr<SQLPhysicalPlanCache> _pqp_cache;
 
   const std::string _select_query_a = "SELECT * FROM table_a";
+  const std::string _select_query_using_join_to_semi_join_optimization_a =
+      "SELECT table_b.a FROM table_a, table_b WHERE table_a.a = table_b.a";
```

Comment:
That's not used (same for the keys constraint on `_table_a`), right? I think there should be tests for the caching behavior, but (similar to the optimizer test), these should mock rules where we explicitly prohibit caching.